### ISSUE TITLE
pydocstyle should not immediately quit when it can't evaluate contents of __all__

### ISSUE THREAD

**Issue by 5j9** [PyCQA/pydocstyle#297]
Create a module with the following content:
```
__all__ = ('a',)
__all__ += ('b',)


a = 1
b = 2
```

Run pydocstyle on it:

```
$ pydocstyle.exe mymodule.py
WARNING: Error in file mymodule.py: Could not evaluate contents of __all__.
That means pydocstyle cannot decide which definitions are
public. Variable __all__ should be present at most once in
each file, in form
`__all__ = ('a_public_function', 'APublicClass', ...)`.
More info on __all__: http://stackoverflow.com/q/44834/. ')
```

I think that even if it can't evaluate `__all__`, it still can and should warn about `D100: Missing docstring in public module`.

**Comment by Nurdok** [comment]
Seems reasonable!

**Comment by Nurdok** [comment]
Fixed by #320.
